main/src/main/scala/sbt/internal/TaskProgress.scala

```diff
@@ -86,20 +86,24 @@ private[sbt] final class TaskProgress(currentRef: ProjectRef) extends ExecutePro
 
   private[this] val console = ConsoleOut.systemOut
   private[this] def readyLog(): Unit = {
```

Comment:
Could this itself just be a lazy val and then you could get rid of the AtomicBoolean?

Reply:
That's a good idea.